src/main_menu.cpp

```diff
@@ -68,10 +68,18 @@ class demo_ui : public cataimgui::window
         void on_resized() override {
             init();
         };
+
+    private:
+        std::shared_ptr<cataimgui::Paragraph> stuff;
 };
 
 demo_ui::demo_ui() : cataimgui::window( _( "ImGui Demo Screen" ) )
 {
+    // char *text = "Some long text that will wrap around nicely. </color> <color_green><color_red>Some red text in the </color>middle.</color>  Some long text that will <color_light_blue_yellow>wrap around nicely.";
```

Comment:
What is the point of mismatched tags being accepted in the first place? Why isn't failing fast better?

Reply:
I haven’t decided that mismatched tags should definitely be accepted, only that they should not cause a crash. For all I know the json already has a mismatched tag or two somewhere.

Reply:
So does your code silently ignore it? Or is there a debug msg?

I thing adding a debugmsg would be the ideal solution. It allows for the bug discovery while not making the game unplayable.

Reply:
Looks like silent ignore:
```
                case color_tag_parse_result::close_color_tag:
                    if( colors.size() > 1 ) {
                        colors.pop_back();
                    }
```

I would add there `else {debugmsg("Encountered closing tag </color> without opening tag. In file X on line Y")}`, if that can be done.